clarity/src/vm/types/serialization.rs

```diff
@@ -782,14 +782,12 @@ impl Value {
                                         expected_type.unwrap(),
                                     ));
                                 }
-                            } else {
-                                if len as u64 != tuple_type.len() {
-                                    // unwrap is safe because of the match condition
-                                    #[allow(clippy::unwrap_used)]
-                                    return Err(SerializationError::DeserializeExpected(
-                                        expected_type.unwrap(),
-                                    ));
-                                }
+                            } else if len as u64 != tuple_type.len() {
```

Comment:
Can we do an explicit `u64::try_from()` conversion of `len` to `u64` so it doesn't truncate or exhibit UB?

Reply:
We could but this len is actually infallible as its casting UP from u32 to u64, but I will make it explicit so its clear using the from instead of "as"